### ISSUE TITLE
the problem with Date

### ISSUE THREAD

**Issue by appbeijing** [gcanti/tcomb-form-native#191]
I hope to set Date a default value,but I can't pass Verification before I change the value by hand.

the code
 `var Person = t.struct({
  name: t.String,
  surname: t.maybe(t.String),
  age: t.Number,
  birth:t.maybe(t.Date),
});

var value={
    name: 'Giulio',
    surname: 'Canti',
    age:"55",
    birth:"Tue Jul 5 2016 10:04:52 GMT+0800 (中国标准时间)"
  };`

Date is setted  optional, but it's still required.

**Comment by willywos** [comment]
haha just noticed this also, have you discovered any workarounds for not making it required?

**Comment by gcanti** [comment]
Hi all, is this related to this PR? https://github.com/gcanti/tcomb-form-native/pull/188

**Comment by appbeijing** [comment]
No  It's not related to PR[#188,](https://github.com/gcanti/tcomb-form-native/pull/188),
this problem has two aspects.
1、even you set Date optional,but it's required, in other words, t.maybe of Date did not work.
2、you have to changed the Date by hand before you submit even Date already has a default value.

PR[#188,](https://github.com/gcanti/tcomb-form-native/pull/188) is about the the value of Date which is always current time. so they are not same problem. after I update the code,I find the the function of TimePickerAndroid was not be changed.

`         .then(function (time) {
                if (time.action !== TimePickerAndroid.dismissedAction) {
                  **var newTime = new Date();**
                  newTime.setHours(time.hour);
                  newTime.setMinutes(time.minute);
                  locals.onChange(newTime);
                }
              });`


`         .then(function (time) {
                if (time.action !== TimePickerAndroid.dismissedAction) {
                  **var newTime = locals.value || new Date();**
                  newTime.setHours(time.hour);
                  newTime.setMinutes(time.minute);
                  locals.onChange(newTime);
                }
              });`

**Comment by alvaromb** [comment]
Hello @appbeijing!

I don't really get what's the issue here, can you give me a more detailed explanation?

**Comment by appbeijing** [comment]
it's always required no matter what you set
@alvaromb

**Comment by ramsestom** [comment]
Don't know if it is exactly the same problem but for me, on android (not tested on iOS), if I define an initial value for a date, I am then unable to change it or event validate the form...

**Comment by narayananvenk** [comment]
I had this issue when I was using DatePicker. The workaround that I used was to override the transformer for DatePicker.

The default transformer:

DatePicker.transformer = {
  format: value => Nil.is(value) ? new Date() : value,
  parse: value => value
};

We can override it in our component as:

DatePicker.transformer = {
  format: value => value,
  parse: value => value
};

Note that I did this in a DatePicker factory that I use, so haven't tested this on the default DatePicker that this module provides.
